src/main/java/sparksoniq/spark/SparkSessionManager.java

```diff
@@ -100,24 +101,30 @@ private void initialize() {
     }
 
     private void initializeKryoSerialization() {
-        this.configuration.set("spark.serializer", "org.apache.spark.serializer.KryoSerializer");
-        Class<?>[] serializedClasses = new Class[] {
-            Item.class,
-            ArrayItem.class,
-            ObjectItem.class,
-            StringItem.class,
-            IntegerItem.class,
-            DoubleItem.class,
-            DecimalItem.class,
-            NullItem.class,
-            BooleanItem.class,
-            DynamicContext.class,
-            FlworTuple.class,
-            FlworKey.class,
-            RuntimeIterator.class,
-            RuntimeTupleIterator.class };
-
-        this.configuration.registerKryoClasses(serializedClasses);
+        String serializer = null;
+        try {
+            serializer = this.configuration.get("spark.serializer");
+        } catch (NoSuchElementException e) {
```

Comment:
Couldn't we write this without resorting to exception catching?

Reply:
Yes, very good point, fixed now.